{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Water Safety App for Dogs üê∂\n",
    "\n",
    "This notebook evaluates water safety for swimming based on chlorophyll and turbidity levels. It checks whether the water is safe for dogs to bathe in, using real-time data from an API (AquaSat and geocoding services)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import folium\n",
    "from flask import Flask, render_template\n",
    "import plotly.graph_objects as go"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import tkinter as tk\n",
    "from tkinter import ttk\n",
    "from tkinter import messagebox\n",
    "from ttkbootstrap import Style\n",
    "from PIL import Image, ImageTk\n",
    "import ee\n",
    "from datetime import datetime, timedelta \n",
    "from opencage.geocoder import OpenCageGeocode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/miniconda3/lib/python3.12/tkinter/__init__.py\", line 1968, in __call__\n",
      "    return self.func(*args)\n",
      "           ^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 92, in on_search_button_click\n",
      "    coordinates = get_coordinates(address)\n",
      "                  ^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/var/folders/rk/7cxcvzcs2pg_yf9qdjfd6byh0000gn/T/ipykernel_1546/3650246392.py\", line 7, in get_coordinates\n",
      "    response = requests.get(url)\n",
      "               ^^^^^^^^\n",
      "NameError: name 'requests' is not defined\n"
     ]
    }
   ],
   "source": [
    "# App Using Tkinter \n",
    "\n",
    "# Function to get coordinates from OpenCage API\n",
    "def get_coordinates(address):\n",
    "    api_key = os.getenv('OpenCage_APIKEY')\n",
    "    url = f\"https://api.opencagedata.com/geocode/v1/json?q={address}&key={api_key}\"\n",
    "    response = requests.get(url)\n",
    "    if response.status_code == 200:\n",
    "        result = response.json()\n",
    "        if result['results']:\n",
    "            lat = result['results'][0]['geometry']['lat']\n",
    "            lng = result['results'][0]['geometry']['lng']\n",
    "            return lat, lng\n",
    "        else:\n",
    "            messagebox.showerror(\"Error\", \"Address not found.\")\n",
    "            return None\n",
    "    else:\n",
    "        messagebox.showerror(\"Error\", f\"Error: {response.status_code}\")\n",
    "        return None\n",
    "\n",
    "# Function to fetch water quality data from Earth Engine and AquaSat\n",
    "def fetch_water_quality(lat, lon):\n",
    "    # Initialize Earth Engine\n",
    "    ee.Initialize()\n",
    "\n",
    "    # Define a point of interest\n",
    "    point = ee.Geometry.Point([lon, lat])\n",
    "    \n",
    "    # Define time range for the last 7 days\n",
    "    end_date = datetime.utcnow()\n",
    "    start_date = end_date - timedelta(days=7)\n",
    "    \n",
    "    # Fetch Sentinel-3 OLCI data for the point within the last 7 days\n",
    "    sentinel3 = ee.ImageCollection(\"COPERNICUS/S3/OLCI\") \\\n",
    "        .filterBounds(point) \\\n",
    "        .filterDate(start_date.strftime('%Y-%m-%d'), end_date.strftime('%Y-%m-%d')) \\\n",
    "        .select(['Oa02_radiance', 'Oa03_radiance', 'Oa04_radiance', 'Oa05_radiance', 'Oa08_radiance'])\n",
    "    \n",
    "    # Check if any images exist\n",
    "    if sentinel3.size().getInfo() == 0:\n",
    "        return None, \"No Sentinel-3 data available for the given location in the last 7 days.\"\n",
    "    \n",
    "    # Compute mean radiance values for each band over the time range\n",
    "    thresholds = {'Oa02_radiance': 0.05, 'Oa03_radiance': 0.05, 'Oa04_radiance': 0.05, 'Oa05_radiance': 0.05, 'Oa08_radiance': 0.05}\n",
    "    flagged_dates = []\n",
    "\n",
    "    # Loop through images\n",
    "    images = sentinel3.toList(sentinel3.size())\n",
    "    for i in range(images.size().getInfo()):\n",
    "        image = ee.Image(images.get(i))\n",
    "        mean_values = image.reduceRegion(\n",
    "            reducer=ee.Reducer.mean(),\n",
    "            geometry=point,\n",
    "            scale=300\n",
    "        ).getInfo()\n",
    "\n",
    "        # Check each band against the thresholds\n",
    "        for band, threshold in thresholds.items():\n",
    "            if band in mean_values and mean_values[band] is not None and mean_values[band] > threshold:\n",
    "                date = image.date().format('YYYY-MM-dd').getInfo()\n",
    "                flagged_dates.append((date, band))\n",
    "                break  # If one band is flagged, no need to check others\n",
    "\n",
    "    # Return flagged dates and bands if any\n",
    "    if flagged_dates:\n",
    "        return flagged_dates, None\n",
    "    else:\n",
    "        return None, \"No flags detected in the last 7 days.\"\n",
    "\n",
    "# Function to fetch nearby beaches using Google Places API\n",
    "def fetch_nearby_beaches(lat, lon, api_key):\n",
    "    url = f\"https://maps.googleapis.com/maps/api/place/nearbysearch/json?location={lat},{lon}&radius=5000&keyword=beach&key={api_key}\"\n",
    "    response = requests.get(url)\n",
    "    places = response.json().get(\"results\", [])\n",
    "\n",
    "    nearby_beaches = []\n",
    "    for place in places[:5]:  # Limit to 5 results\n",
    "        name = place.get('name')\n",
    "        vicinity = place.get('vicinity', 'Unknown location')\n",
    "        nearby_beaches.append(f\"{name} - {vicinity}\")\n",
    "    \n",
    "    return nearby_beaches\n",
    "\n",
    "# Function to handle button click\n",
    "def on_search_button_click():\n",
    "    address = entry.get()  # Get the address from the input field\n",
    "    if not address:\n",
    "        messagebox.showerror(\"Input Error\", \"Please enter an address.\")\n",
    "        return\n",
    "    \n",
    "    # Get coordinates from address\n",
    "    coordinates = get_coordinates(address)\n",
    "    if coordinates:\n",
    "        lat, lng = coordinates\n",
    "        result_label.config(text=f\"Coordinates: Latitude: {lat}, Longitude: {lng}\")\n",
    "        \n",
    "        # Fetch water quality data\n",
    "        try:\n",
    "            flagged_data, message = fetch_water_quality(lat, lng)\n",
    "            if flagged_data:\n",
    "                flagged_list = \"\\n\".join([f\"{date} - {band}\" for date, band in flagged_data])\n",
    "                messagebox.showwarning(\"Water Quality Alert\", f\"Wait! üê∂ \\n\\n Water near your location may have increased vegetation and murky conditions. \\n\\n Check your local health department's website for updates on water quality near you.\")\n",
    "                \n",
    "                # Fetch nearby beaches\n",
    "                google_api_key = \"google_api_key\"  \n",
    "                nearby_beaches = fetch_nearby_beaches(lat, lng, google_api_key)\n",
    "                if nearby_beaches:\n",
    "                    messagebox.showinfo(\"Nearby Beaches\", \" \\n\".join(nearby_beaches))\n",
    "                else:\n",
    "                    messagebox.showinfo(\"Nearby Beaches\", \"No nearby beaches found.\")\n",
    "            else:\n",
    "                messagebox.showinfo(\"Water Quality\", \"Water quality is safe in the last 7 days.\")\n",
    "            root.destroy()  # Stop execution by closing the app\n",
    "        except Exception as e:\n",
    "            messagebox.showerror(\"Error\", f\"Error fetching water quality: {e}\")\n",
    "\n",
    "# Set up the Tkinter window with ttkbootstrap\n",
    "root = tk.Tk()\n",
    "style = Style('superhero')  # You can change the theme here\n",
    "style.master = root\n",
    "\n",
    "root.title(\"DjurSafe\")\n",
    "root.geometry(\"400x600\")\n",
    "\n",
    "# Splash screen elements\n",
    "title_label = ttk.Label(root, text=\"DjurSafe\", font=(\"Helvetica\", 45, \"bold\"), anchor=\"center\", background=style.lookup(\"TFrame\", \"background\"))\n",
    "title_label.pack(pady=10)\n",
    "\n",
    "description_label = ttk.Label(\n",
    "    root, \n",
    "    text=\"Out and about? \\n\\n Enter an address near you for water quality updates in the last 7 days:\", \n",
    "    wraplength=350, \n",
    "    font=(\"Helvetica\", 25),\n",
    "    anchor=\"center\",\n",
    "    justify=\"center\",\n",
    "    background=style.lookup(\"TFrame\", \"background\")\n",
    ")\n",
    "description_label.pack(pady=10)\n",
    "\n",
    "# Add dog icon\n",
    "try:\n",
    "    dog_image = Image.open(\"/Users/jt/Downloads/dog.png\")\n",
    "    dog_image = dog_image.resize((200, 200))  # Resize if necessary\n",
    "    dog_photo = ImageTk.PhotoImage(dog_image)\n",
    "    dog_label = ttk.Label(root, image=dog_photo)\n",
    "    dog_label.pack(pady=10)\n",
    "except Exception as e:\n",
    "    messagebox.showerror(\"Error\", f\"Failed to load image: {e}\")\n",
    "\n",
    "# Input field for the address\n",
    "entry = ttk.Entry(root, width=50)\n",
    "entry.pack(pady=10)\n",
    "\n",
    "# Search button\n",
    "search_button = ttk.Button(root, text=\"Search\", command=on_search_button_click)\n",
    "search_button.pack(pady=10)\n",
    "\n",
    "# Result label to show coordinates and water quality status\n",
    "result_label = ttk.Label(root, text=\"Coordinates and water quality will be displayed here.\", wraplength=350)\n",
    "result_label.pack(pady=10)\n",
    "\n",
    "# Main loop\n",
    "root.mainloop()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
